lib/fizzy/parser.cpp

```diff
@@ -261,42 +286,46 @@ Module parse(bytes_view input)
     {
         const auto id = static_cast<SectionId>(*it++);
         uint32_t size;
-        std::tie(size, it) = leb128u_decode<uint32_t>(it);
-        const auto expected_end_pos = it + size;
+        std::tie(size, it) = leb128u_decode<uint32_t>(it, input.end());
```

Comment:
Not sure if it's worth to create `input_end` variable instead of repeated `end()` calls

Reply:
Rather not.